src/stdio.rs

```diff
@@ -95,13 +95,19 @@ impl AsFd for PipeStdin {
     }
 }
 
+// Implementation based on the implementation in tokio_util::compat
 impl AsyncRead for PipeStdin {
```

Comment:
I believe the compat is due to using tokio's child process spawn, which returns tokio streams, instead of the PipeStdin/PipeStdout defined in this library.

Reply:
Oh I misunderstood. So you changed `Frontend::run` to accept `Async{Read,Write}` from futures instead of tokio, and tokio `Process` used by the test does not implement `Async*` from futures.